src/util/cache/cache.h

```diff
@@ -0,0 +1,84 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#pragma once
+
+#include "../priority_queue.h"
+#include "rax.h"
+#include "pthread.h"
+
+#define HASH_KEY_LENGTH 8
+
+typedef unsigned long long hash_key_t;
+typedef void (*cacheValueFreeFunc)(void *);
+
+/**
+ * @brief  Struct for holding cache data.
+ */
+typedef struct CacheData {
+	hash_key_t hashKey;             // CacheData key - 64 bit hashed key.
+	void *value;                    // Value to be stored in cache.
+	cacheValueFreeFunc freeFunc;    // function to free the stored value.
```

Comment:
Within a given cache I think all items are of the same type as such the `freeFunc` should be the same for all,
no need to keep it at the value level.

Reply:
you will need to have a pointer to it anyway